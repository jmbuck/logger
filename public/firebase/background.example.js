// Initialize Firebase
var config = {
    apiKey: "EXAMPLE",
    authDomain: "EXAMPLE",
    databaseURL: "EXAMPLE",
    projectId: "EXAMPLE",
    storageBucket: "EXAMPLE",
    messagingSenderId: "EXAMPLE"
};
firebase.initializeApp(config);

/**
 * initApp handles setting up the Firebase context and registering
 * callbacks for the auth status.
 *
 * The core initialization is in firebase.App - this is the glue class
 * which stores configuration. We provide an app name here to allow
 * distinguishing multiple app instances.
 *
 * This method also registers a listener with firebase.auth().onAuthStateChanged.
 * This listener is called when the user is signed in or out, and that
 * is where we update the UI.
 *
 * When signed in, we also authenticate to the Firebase Realtime Database.
 */
function initApp() {
    // Listen for auth state changes.
    firebase.auth().onAuthStateChanged(function(user) {
        console.log('User state change detected from the Background script of the Chrome Extension:', user);
    });
}

window.onload = function() {
    initApp();
};