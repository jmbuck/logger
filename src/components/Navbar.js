import React, { Component } from 'react'
import '../css/common.css'
import logout from '../img/logout.svg'
import settings from '../img/settings.svg'
import {auth} from '../rebase'

class Navbar extends Component {
  render() {
    return (
    <div className="Navbar">
        <div className="NavList">
            <div className="Options">
                <div className="Settings">
                    <img src={settings} onClick={() => this.props.history.push("/preferences")} />
                </div>
                <div className="Logout">
                    <img src={logout} onClick={() => auth.signOut().then(() => {
                        this.setState({ data: {} });
                        this.props.history.push('/login')
                    })}/>
                </div>
            </div>
        </div>
        <div className="Logger-Title" >
            <span onClick={() => this.props.history.push('/data')}>Logger</span>
        </div>
    </div>
    )
  }
}

export default Navbar