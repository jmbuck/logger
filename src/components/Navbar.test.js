import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Navbar from "./Navbar";
import firebase from "firebase";

Enzyme.configure({ adapter: new Adapter() });

describe("Navbar", () => {
    beforeAll(() => {
        firebase.auth = jest.fn().mockReturnValue({
            onAuthStateChanged: true
        });
    });

    it("should render component", () => {
        const component = Enzyme.shallow(<Navbar />);

        expect(component).toMatchSnapshot();
    });
});

