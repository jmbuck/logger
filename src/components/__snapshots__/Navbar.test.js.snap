// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="NavList"
>
          <div
                    className="Options"
          >
                    <div
                              className="Settings"
                    >
                              <img
                                        onClick={[Function]}
                                        src="settings.svg"
                              />
                    </div>
                    <div
                              className="Logout"
                    >
                              <img
                                        onClick={[Function]}
                                        src="logout.svg"
                              />
                    </div>
          </div>
</div>,
        <div
          className="Logger-Title"
>
          <span
                    onClick={[Function]}
          >
                    Logger
          </span>
</div>,
      ],
      "className": "Navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="Options"
>
            <div
                        className="Settings"
            >
                        <img
                                    onClick={[Function]}
                                    src="settings.svg"
                        />
            </div>
            <div
                        className="Logout"
            >
                        <img
                                    onClick={[Function]}
                                    src="logout.svg"
                        />
            </div>
</div>,
          "className": "NavList",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Settings"
>
                <img
                                onClick={[Function]}
                                src="settings.svg"
                />
</div>,
              <div
                className="Logout"
>
                <img
                                onClick={[Function]}
                                src="logout.svg"
                />
</div>,
            ],
            "className": "Options",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  onClick={[Function]}
                  src="settings.svg"
/>,
                "className": "Settings",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onClick": [Function],
                  "src": "settings.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  onClick={[Function]}
                  src="logout.svg"
/>,
                "className": "Logout",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onClick": [Function],
                  "src": "logout.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            onClick={[Function]}
>
            Logger
</span>,
          "className": "Logger-Title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logger",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logger",
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="NavList"
>
            <div
                        className="Options"
            >
                        <div
                                    className="Settings"
                        >
                                    <img
                                                onClick={[Function]}
                                                src="settings.svg"
                                    />
                        </div>
                        <div
                                    className="Logout"
                        >
                                    <img
                                                onClick={[Function]}
                                                src="logout.svg"
                                    />
                        </div>
            </div>
</div>,
          <div
            className="Logger-Title"
>
            <span
                        onClick={[Function]}
            >
                        Logger
            </span>
</div>,
        ],
        "className": "Navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="Options"
>
              <div
                            className="Settings"
              >
                            <img
                                          onClick={[Function]}
                                          src="settings.svg"
                            />
              </div>
              <div
                            className="Logout"
              >
                            <img
                                          onClick={[Function]}
                                          src="logout.svg"
                            />
              </div>
</div>,
            "className": "NavList",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="Settings"
>
                  <img
                                    onClick={[Function]}
                                    src="settings.svg"
                  />
</div>,
                <div
                  className="Logout"
>
                  <img
                                    onClick={[Function]}
                                    src="logout.svg"
                  />
</div>,
              ],
              "className": "Options",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    onClick={[Function]}
                    src="settings.svg"
/>,
                  "className": "Settings",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "onClick": [Function],
                    "src": "settings.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    onClick={[Function]}
                    src="logout.svg"
/>,
                  "className": "Logout",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "onClick": [Function],
                    "src": "logout.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              onClick={[Function]}
>
              Logger
</span>,
            "className": "Logger-Title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logger",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Logger",
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
